test/runtests.jl

```diff
@@ -51,6 +52,46 @@ Random.seed!(100)
     @test cost2 ≈ cost
 end
 
+@testset "1D Optimal Transport for Convex Cost" begin
+    # Continuous Case
+    μ = Normal(0, 2)
+    ν = Normal(10, 2)
+    c(x, y) = abs(x - y)
+    γ = optimal_transport_plan(c, μ, ν)
+
+    @test optimal_transport_cost(c, μ, ν) ≈ 10 atol = 1e-5
+    @test optimal_transport_cost(c, μ, ν; plan=γ) ≈ 10 atol = 1e-5
+
+    # Discrete Case
+    n, m = 100, 150
+
+    μ = rand(n)
+    ν = rand(m) .+ 0.5
+    μ_n = rand(n)
+    ν_m = rand(m)
+    μ_n = μ_n / sum(μ_n)
+    ν_m = ν_m / sum(ν_m)
+
+    μ = DiscreteNonParametric(μ, μ_n)
+    ν = DiscreteNonParametric(ν, ν_m)
```

Comment:
This is confusing for a reader and also for the compiler sometimes: one should not reassign different types to existing variable names. Better use `\mu_support` for the support.